File: 1630.等差子数组.py
#
# @lc app=leetcode.cn id=1630 lang=python3
#
# [1630] 等差子数组
#

# @lc code=start
class Solution:
    def checkArithmeticSubarrays(self, nums: List[int], l: List[int], r: List[int]) -> List[bool]:
        def isArithmetic(nums):
            n = len(nums)
            if n <= 2:
                return True
            diff = nums[1] - nums[0]
            for i in range(2, n):
                if nums[i] - nums[i-1] != diff:
                    return False
            return True
        m = len(l)
        res = [False] * m
        idx = 0
        for i, j in zip(l, r):
            tmp = sorted(nums[i:j+1])
            res[idx] = isArithmetic(tmp)
            idx += 1
        return res

# @lc code=end

File: 1630.等差子数组.go
import "sort"

/*
 * @lc app=leetcode.cn id=1630 lang=golang
 *
 * [1630] 等差子数组
 */

// @lc code=start
func checkArithmeticSubarrays(nums []int, l []int, r []int) []bool {
	m = len(l)
	res = make([]bool, m)
	for i := 0; i < m; i++ {
		res[i] = false
	}
	for i := 0; i < m; i++ {
		res[i] = check(nums[l[i] : r[i]+1])
	}
	return res

}

func check(nums []int) bool {
	sort.Ints(nums)
	diff := nums[1] - nums[0]
	for i := 2; i < len(nums); i++ {
		if nums[i]-nums[i-1] != diff {
			return false
		}
	}
	return true
}

// @lc code=end

